### ISSUE TITLE
login == Network error: Failed to fetch after docker-compose and docker exec

### ISSUE THREAD

**Issue by jodumont** [erxes/erxes#801]
Hi;

I ran the first 5steps I found here (https://github.com/erxes/erxes/wiki/Use-erxes-with-Docker)

than when I'm going to the localhost:3000 and try 
(username: admin@erxes.io , password: erxes)
I receive this error : Network error: Failed to fetch

If I run the command docker exec -it erxes-api yarn initProject again
Mongo say the email is already in the database

How I could corect this ?

**Comment by shinebayar-g** [comment]
Hi, there could only few cases docker instrument will fail.
First of all there could be port conflict on your system. If your system already using those ports, erxes cannot bind on those ports. (3000, 3100, 3200, 3300, 6379, 27017) Please check if all erxes containers up by typing `docker ps` command. It should show all 6 containers are up and running like this.

If it's not you can check dead containers by typing `docker ps -a`. In this case you should some containers with (exit) status. Find those container's name and check its logs by typing `docker logs <container name here>` command. Example, let's say mongodb is not up. You can check what's wrong with it running `docker logs mongo` command.
If it's says cannot bind on port or something like that, you have most likely found the problem.
Please turn off port conflicting services and try again with. `docker-compose start` (it restarts dead containers only, not alive containers) or `docker-compose down, docker-compose up -d`.

If all containers shows up and running. You can check your browser's console to find out what's happening there. (F12 on chrome). There will be bunch of red errors on console tab. If you see them, you should screenshot them here. Thank you.

**Comment by jodumont** [comment]
Hi :)

It's nice to see great project with reactive people

thanks for your response
so everything works on the docker side

Here a screenshot of Chrome debugger
but basicly GraphQL is not connecting
`WebSocket connection to 'ws://localhost:3300/subscriptions' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED`

here the line 545 of client.js `this.client = new this.wsImpl(this.url, protocol_1.GRAPHQL_WS);`

![image](https://user-images.githubusercontent.com/5204724/51020998-86054400-1580-11e9-9041-0373343d6aa7.png)

PS: should I mention then I'm using the IP like 199.19.99.9:3000 and not localhost:3000 to access it ?

**Comment by shinebayar-g** [comment]
Glad to hear that. Let's debug it. Are you running erxes on your local machine or remote machine?
And are you using domain or using just ports?

**Comment by jodumont** [comment]
on a dedicated for test remote machine
for now I'm using a ip:port but I could easily attach a domain name to it

**Comment by shinebayar-g** [comment]
I see, in that case you should put your IPs in your environment variables configuration. Example: If you take a look at docker-compose.yml file's this part:
```YAML
erxes:
    image: erxes/erxes:docker-latest
    container_name: erxes
    environment:
      REACT_APP_CDN_HOST: http://localhost:3200 (This should be changed to your server's IP, example 173.10.59.22 etc)
      REACT_APP_CDN_HOST_API: http://localhost:3100 (This too)
      REACT_APP_API_URL: http://localhost:3300
      REACT_APP_API_SUBSCRIPTION_URL: ws://localhost:3300/subscriptions
      NGINX_HOST: localhost
    ports:
      - "3000:80"
    networks:
      - erxes-net
```

This settings should be changed on all erxes components. Because by default this configuration assumes you're running it on your local machine, thus results calling localhost:3300 which is invalid on your case. I knew this should be documented on our wiki. I hope this will fix your issue.

**Comment by jodumont** [comment]
just be sure;

I replace all localhost by the IP ?
or just for the erxes container ?

because I did `sed -i s/$IP/localhost/g docker-compose.yml`
than `docker-compose up -d`
and I still have the same error

**Comment by shinebayar-g** [comment]
> This settings should be changed on all erxes components. 
I thought I mentioned that there. Sorry if that sentence didn't make sense to you. (I am not native speaker so)

**Comment by jodumont** [comment]
don't worry;
I'm a french canadian so, I'm not native too
and yes your example was clear, I just need more coffee :)

so now it only a matter of or CORS policy

Access to fetch at 'http://199.19.99.9:3300/graphql' from origin 'http://199.19.99.9' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by shinebayar-g** [comment]
Great, that errors should be fixed once I you add new environment variables to your config.

**Comment by jodumont** [comment]
> once you add new environment variables

you lost me on this one

**Comment by jodumont** [comment]
saddly I receive an error saying it should be : instead of = for these lines
```
  in "./docker-compose.yml", line 39, column 7
could not find expected ':'
  in "./docker-compose.yml", line 40, column 7
```

but when I replace them by
```
  USE_REPLICATION: false
  HTTPS: false
  WIDGETS_DOMAIN=http://localhost:3200
```
it result this error
```
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.erxes-api.environment.USE_REPLICATION contains false, which is an invalid type, it should be a string, number, or a null

```

**Comment by shinebayar-g** [comment]
Oh god, I made a typo mistake it shouldn't be `=` it should be `:`
```YAML
  USE_REPLICATION: false
  HTTPS: false
  WIDGETS_DOMAIN: http://localhost:3200
```
try again.

**Comment by jodumont** [comment]
NOPE :P

ERROR: The Compose file './docker-compose.yml' is invalid because:
services.erxes-api.environment.USE_REPLICATION contains false, which is an invalid type, it should be a string, number, or a null

**Comment by jodumont** [comment]
if I remove USE_REPLICATION than I have the same error on the HTTPS line

`services.erxes-api.environment.HTTPS contains false, which is an invalid type, it should be a string, number, or a null
`

**Comment by jodumont** [comment]
it run if

```
  USE_REPLICATION: "false"
  HTTPS: "false"

```

**Comment by shinebayar-g** [comment]
Alright, I'm finally able to test it with you. Yes god damn, it should be "".

**Comment by jodumont** [comment]
oh BTW at the end I'll put the nginx proxy in front, so if you think it is just easier to put the nginx proxy than keeping everything with localhost, please tell me :)

**Comment by shinebayar-g** [comment]
It should be fine without nginx at the moment, btw if you can test it on your local computer in the meantime.

**Comment by jodumont** [comment]
> on your local computer

it is on it's way

**Comment by shinebayar-g** [comment]
Aight, I just tested in on my remote server.
Step 1. Copy paste docker-compose.yml file
Step 2. replace localhost -> my-server-ip
Step 3. `docker-compose up -d`
Step 4. `docker exec -it erxes-api yarn initProject`
Step 5. Browse to my-server-ip:3000
Profit :P 

It's all working for my end.

**Comment by jodumont** [comment]
on my localhost it work
on the remote server I still have the CORS error


PS: don't forget to correct the [wiki](https://github.com/erxes/erxes/wiki/Use-erxes-with-Docker)
```
      USE_REPLICATION: "false"
      HTTPS: "false"
```

**Comment by shinebayar-g** [comment]
Thanks for reminder, i thought i updated that. Did you try browser's
incognito mode?

**Comment by shinebayar-g** [comment]
@jodumont  I'm so sorry to bothering you with this issue. I hope we worked it out. If you encounter any problem please feel free to raise an another issue. We truly care every single user, please let me know your thoughts about project. I afraid you'd hit the wall again while wandering through our application's settings etc. It's not greatly documented at the moment, if you want and serious to use our app you can have private session with me.
